plugins/cliconf/iosxr.py

```diff
@@ -310,28 +335,47 @@ def get(
         )
 
     def commit(self, comment=None, label=None, replace=None):
+        """Implements commit functionality of config module
+        and commit confirmed functionality of cliconf module
+
+        Args:
+            comment (str, optional): commit comment. Defaults to None.
+            label (str, optional): commit label. Defaults to None.
+            replace (bool, optional): Flag to replace commit. Defaults to None.
+        """
         cmd_obj = {}
         if replace:
             cmd_obj["command"] = "commit replace"
             cmd_obj[
                 "prompt"
             ] = "This commit will replace or remove the entire running configuration"
             cmd_obj["answer"] = "yes"
-        elif self.get_option("commit_confirmed") and self.get_option(
+
+        elif self.get_option("commit_confirmed") or self.get_option(
```

Comment:
If `commit_confirmed` is disabled but `commit_confirmed_timeout` is set, would it still issue a commit confirmed with a timeout? If so, I don't think that's correct. We can have a `commit_confirmed_timeout` globally set, but might want `commit_confirmed` to be conditionally turned off via task vars for a subset of tasks in a play.

Reply:
added the change, without and'ng the commit confirmed attribute with other commit confirmed attributes, it can make the user generate the wrong command and fail if they use label/comment/timeout without commit_confirmed.
Thanks for the suggestion